File: pantry-tracker-backend/database_operations/emptyGroceryList.py
from extensions import db
from sqlalchemy import text
from flask import Blueprint, jsonify

emptyGroceryList_bp = Blueprint('emptyGroceryList', __name__)

@emptyGroceryList_bp.route('/empty-grocery-list', methods=['POST'])

def emptyGroceryList():
    try:
        db.session.execute(text('UPDATE my_grocery_list SET quantity = 0 WHERE quantity > 0'))
        db.session.commit()
        return jsonify({'success': True, 'message': 'Grocery List Emptied Successfully'}), 200
    except Exception as e:
        return jsonify({'success': False, 'error': str(e)}), 500

File: pantry-tracker-frontend/src/supabase_db_operations/emptyGroceryList.js
import supabaseClient from '../utils/SupabaseClient';

const emptyGroceryList = async (userID) => {
    try {
        console.log("running successfully!");
        const { error: emptyGroceryListError } = await supabaseClient
            .from('user_ingredients')
            .update({ 'gl_quantity': 0 })
            .eq('user_id', userID)

        if (emptyGroceryListError) {
            throw emptyGroceryListError;
        }
        console.log("returning true");
        return { success: true };
    }
    catch (error) {
        console.error(error);
        return { 'error': error.message };
    }
};

export default emptyGroceryList;